lms/djangoapps/certificates/api.py

```diff
@@ -467,8 +467,8 @@ def _certificate_download_url(user_id, course_id, user_certificate=None):
         except GeneratedCertificate.DoesNotExist:
             log.critical(
                 u'Unable to lookup certificate\n'
-                u'user id: %d\n'
```

Comment:
For an anonymous user, this breaks because the ID is `None`.  This change isn't strictly necessary for tests to pass, but the bug hampered debugging and could happen in prod logging.